packages/core/package.json

```diff
@@ -54,6 +54,7 @@
         "@apify/log": "^1.2.0",
         "@apify/pseudo_url": "^1.1.0",
         "@apify/timeout": "^0.2.1",
+        "@crawlee/browser-pool": "^3.0.0",
```

Comment:
i dont have time to read through the comments right now, but i already said this is wrong, we need to remove the dependency, not declare it

i also dont fully understand what your problem with the user agents is, we should backport things from apify-js - there it works correctly and there you can see what is released to stable v2.